### PR TITLE
Add edition selection for integration runtime

### PR DESCRIPTION
**Pull request by chinadragon0515**
## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [ ] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [ ] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [NA] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [NA] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by maddieclayton** [comment]
@chinadragon0515  Please update the [changelog](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/DataFactories/Commands.DataFactoryV2/Changelog.md), [regenerate the help files](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md), and add tests to cover these new parameters.

**Comment by chinadragon0515** [comment]
I have updated the change log, regenerate help file and add test for edition properties.

One comment is for generate the help file, prefer some update on the guide,
1. there is sample for access psd1 file from debug folder, then which help folder we should refer too?
2. Assume help folder is from debug folder, then we will need to copy generate md file from help folder to product folder.
3. Additional blank line needed is not documented per comments from peers.
4. Maybe can we make this as part of build process?

**Comment by chinadragon0515** [comment]
SDK was merged yesterday, when it will be available for pwoershell build? The build failed now is because it cannot find classes added to the SDK.

**Comment by maddieclayton** [comment]
@chinadragon0515 
1) I will update the help-generation file with an example of the which help folder to use:
 (src/ResourceManager/Profile/Commands.Profile/help).
2) No need to copy as you should update the files in the module folder and not the debug folder.
3) We are adding a build-time check to ensure the extra line is included.  Additionally, we have filed a bug with platyPS which will hopefully be resolved soon.
4) Adding this to the build process would add a lot of time to the build, as regenerating every file takes a long time.  We can think about other ways to make this process easier.

To get the new SDK version, please bump the version in the csproj and the packages files for both the module and test:
https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/DataFactories/Commands.DataFactoryV2/Commands.DataFactoryV2.csproj
https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/DataFactories/Commands.DataFactoryV2/packages.config
https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/Commands.DataFactoryV2.Test.csproj
https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/packages.config

**Comment by chinadragon0515** [comment]
I make changes to the SDK, but SDK version is not bumped, the build will need a updated sdk nupkg to do the build, and version is same as before, what should I do?

**Comment by maddieclayton** [comment]
@chinadragon0515 To get the changes to the SDK, you will need to publish a new version.

**Comment by chinadragon0515** [comment]
Per email exchange with ADF team, we will test with private build, and ".Net SDK will be published and that local nuget will be removed by Hermine Vermishyan  or Dean Talley  once we publish our SDK to nuget.org."

**Comment by cormacpayne** [comment]
@chinadragon0515 add the "Do Not Merge" label until the SDK is published to NuGet. It looks like you were able to create a signed NuGet package containing your changes using the Jenkins job. You can place that package into the `tools/LocalFeed` folder for local testing, but this cannot be checked into the repository.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 [hunk 1]
```diff
@@ -113,6 +113,7 @@ function Test-SsisAzure-IntegrationRuntime
             -CatalogPricingTier 'S1' `
             -MaxParallelExecutionsPerNode 1 `
             -LicenseType LicenseIncluded `
+            -Edition Enterprise `
```
Comment (maddieclayton) [inline src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: Please add the SetupScriptContainerSasUri parameter as well, and add an assert to ensure they are set properly.

Comment (chinadragon0515) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: I add SetupScriptContainerSasUri , but did add assert as this case is not enabled now, per discussion with ADF team, they will help to enable the case, there is a task to track this, and it is same case for License Type

**Comment by chinadragon0515** [comment]
@maddieclayton Can you help to review any let me know if there is any more comments?
Per discussion with SDF team, SDK will be published around Feb. 13 (Feb. 14 China time), and as this Chinese New Year holiday will start Feb. 14, @hvermis will help to remove loolfeed and merge the PR.

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 (hunk 1 above)
Comment (maddieclayton) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: If the case is not enabled now, why is the user allowed to add this parameter?  I would think you don't want to merge this change until the case is enabled.

Comment (chinadragon0515) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: No, I mean test case is not enabled, the feature is enabled,but the test case is not enabled yet.
Refer to this issue for detail, https://github.com/Azure/azure-powershell/issues/5327

Can you let me know if you have any more comments?

**Comment by chinadragon0515** [comment]
Test is not enabled yet, it will be enabled via Issue is #5327

File: src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1 (hunk 1 above)
Comment (maddieclayton) [reply src/ResourceManager/DataFactories/Commands.DataFactoryV2.Test/ScenarioTests/IntegrationRuntimeTests.ps1]: This should be good to merge once the nuget package is published.

**Comment by MiYanni** [comment]
@hvermis Any update on publishing the SDK to NuGet?

**Comment by hvermis** [comment]
@MiYanni I am working on publishing it today as planned, and will update this branch as soon it's done.

**Comment by hvermis** [comment]
@MiYanni I just pushed the nuget version update, this PR can be merged now.

**Comment by MiYanni** [comment]
@chinadragon0515 @maddieclayton @hvermis Is this PR still blocked by #5327?

**Comment by hvermis** [comment]
@MiYanni @maddieclayton My understanding is Maddie agreed to merge it and have test resolved separately.

**Comment by maddieclayton** [comment]
on demand here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/480/

**Comment by hvermis** [comment]
@maddieclayton @MiYanni All checks and builds including on demand have passed on this PR, please merge.

**Pull request merged**